preupg/utils.py

```diff
@@ -862,3 +848,74 @@ def run_generate(self, xml_file, html_file, old_style=False):
         return ret_val
 
 
+class ModulSetUtils(object):
+
+    @staticmethod
+    def check_property(ini_file_path, key, section):
+        """
+        Checks for key in section if key exists, returns it else raise
+        an exception
+
+        @param {string} ini_file_path
+        @param {string} key
+        @param {string} section
+        @return {string} - key value
+        @throws {EnvironmentError} - if key of section doesn't exist
+        """
+        if not ConfigHelper.has_ini_file_section(ini_file_path, section):
+            raise EnvironmentError("Section {0} is missing inside {1} file is"
+                                   .format(section, ini_file_path))
+        key_value = ConfigHelper.get_preupg_config_file(ini_file_path,
+                                                        key, section)
+        if not key_value:
+            raise EnvironmentError("Key {0} inside {1} section\n"
+                                   "is missing in {2}"
+                                   .format(key, section, ini_file_path))
+        return key_value
+
+    @staticmethod
+    def get_assessment_version(module_set_path):
+        """
+        Check if properties.ini file exists in dir_name and also
+        validate the file content. From the content gets src/dst version of OS
+        and return them as list.
+
+        @param {string} dir_name - eg. /root/preupgrage/modul_dir
+        @param {string} base_dir
+        @return {list}
+        @throws {EnvironmentError} - when file doesn't exist or content is
+                                     incorrect
+        """
+        src_version_key = "srcMajorVersion"
+        dst_version_key = "dstMajorVersion"
+        section_name = "preupgrade-assistant-modules"
+        if not os.path.isdir(module_set_path):
+            module_set_path = os.path.join(settings.source_dir,
+                                           module_set_path)
+        properties_file = os.path.join(module_set_path,
+                                       settings.properties_ini)
+        if not os.path.isfile(properties_file):
+            raise EnvironmentError("{0} file not found inside {1} directory"
+                                   .format(settings.properties_ini,
+                                           os.path.dirname(properties_file)))
```

Comment:
Fixed in https://github.com/upgrades-migrations/preupgrade-assistant/pull/233/commits/107a550a8ec5aeb8ddcbd5e59fef0d6a99e3e0a2